jira/client.py

```diff
@@ -2896,7 +2896,7 @@ def search_issues(
         if isinstance(fields, str):
             fields = fields.split(",")
         else:
-            fields = list(fields or [])
+            fields = list(fields or ["*all"])
```

Comment:
```suggestion
            fields = list(["*all"] if fields is None else fields)
```

Let's do an explicit check with `None`, that way you can pass an empty list if you actually don't want everything.

I don''t really like how we call `list()` on a `str`

```python
>>> list("hey")
['h', 'e', 'y']
```

Do you think you can tidy this up further? i.e. `if isinstance(fields, str)`

Reply:
I was wondering if it would be better to change the default value to '*all" as the None value actually represents all.
This way we only need to convert string inputs to list and pass through list inputs.

Reply:
Yep that sounds good too

Reply:
Alright then, I'll send a new PR

Reply:
(feel free to just update this branch with more commits!)